File: app/src/main/java/com/skyd/imomoe/util/dlna/dmc/ICast.java
package com.skyd.imomoe.util.dlna.dmc;

import androidx.annotation.NonNull;

public interface ICast {

    @NonNull
    String getId();

    @NonNull
    String getUri();

    String getName();

    interface ICastVideo extends ICast {

        /**
         * @return video duration, ms
         */
        long getDurationMillSeconds();

        long getSize();

        long getBitrate();
    }

    interface ICastAudio extends ICast {
        /**
         * @return audio duration, ms
         */
        long getDurationMillSeconds();

        long getSize();
    }

    interface ICastImage extends ICast {
        long getSize();
    }

}

File: app/src/main/java/com/skyd/imomoe/util/dlna/dmc/ICast.kt
package com.skyd.imomoe.util.dlna.dmc

interface ICast {
    val id: String
    val uri: String
    val name: String?

    interface ICastVideo : ICast {
        /**
         * @return video duration, ms
         */
        val durationMillSeconds: Long
        val size: Long
        val bitrate: Long
    }

    interface ICastAudio : ICast {
        /**
         * @return audio duration, ms
         */
        val durationMillSeconds: Long
        val size: Long
    }

    interface ICastImage : ICast {
        val size: Long
    }
}